### PR TITLE
Create BlackLinter

### PR DESCRIPTION
**Pull request by cbollsandiego**
The old linter was working with showing us errors in formatting, but would not auto-fix anything. This new workflow should correct the code automatically with the black linter.

### CODE REVIEW

**Comment by cbollsandiego** [comment]
Sorry for the mass commits and Actions runs. Hopefully you didn't get spammed with all the emails I got from failed runs. I had to run the actions a lot of times to get things working correctly but the only way that I could find to do it was to make this pull request and then commit to it again and again.

File: .github/workflows/BlackLinter.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        run: git checkout ${{ github.head_ref }}
+
+      - name: Commit and Push Changes
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "actions@github.com"
+          git add .
+          git commit -m "Auto-fix Python code issues with Black"
+          git push origin ${{ github.head_ref }}
+        env:
```
Comment (audreynaidu) [inline .github/workflows/BlackLinter.yml:48]: nice complete set of git commands

File: .github/workflows/BlackLinter.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+          python-version: "3.11.3"
+
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black flake8
+      
+      - name: Run Black (Auto-Fix Code)
+        run: black .
+
```
Comment (audreynaidu) [inline .github/workflows/BlackLinter.yml:30]: sets up python and dependencies correctly

**Review by audreynaidu — APPROVED** [review]
Looks like the revised version of the linter is working well! Great work. It includes a well structured workflow, logical exclusions, and the automation of committing and pushing.

**Pull request merged**